### PR TITLE
Added a way to disconnect transcriptions

### PR DESCRIPTION
**Pull request by albandum**
## Description

After discussion with a few customers, looks like people might have different transcripts providers accross one company. 
- Added a way to disconnect your own current provider
- If it's the last user using that provider, also remove the nango connection 
- Find default connection becomes linked to workspace/provider as discussed @flvndvd 
Screenshot 2024-05-28 at 14 23 20
Screenshot 2024-05-28 at 14 23 18

### CODE REVIEW

File: front/lib/labs/config.ts [hunk 1]
```diff
@@ -7,11 +8,17 @@ const config = {
   getNangoSecretKey: (): string => {
     return EnvironmentConfig.getEnvVariable("NANGO_SECRET_KEY");
   },
-  getNangoGoogleDriveConnectorId: (): string => {
-    return EnvironmentConfig.getEnvVariable("NANGO_GOOGLE_DRIVE_CONNECTOR_ID");
-  },
-  getNangoGongConnectorId: (): string => {
-    return EnvironmentConfig.getEnvVariable("NANGO_GONG_CONNECTOR_ID");
+  getNangoConnectorIdForProvider: (provider: LabsConnectorProvider): string => {
```
Comment (flvndvd) [inline front/lib/labs/config.ts:11]: In the config let's avoid putting logic like this and do one getter per provider.

File: front/pages/api/w/[wId]/labs/transcripts/default.ts [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 import { Authenticator, getSession } from "@app/lib/auth";
 import { LabsTranscriptsConfigurationResource } from "@app/lib/resources/labs_transcripts_resource";
 import { apiError, withLogging } from "@app/logger/withlogging";
 import type { GetLabsTranscriptsConfigurationResponseBody } from "@app/pages/api/w/[wId]/labs/transcripts";
+import { acceptableTranscriptProvidersCodec } from "@app/pages/api/w/[wId]/labs/transcripts";
 
 const PROVIDERS_WITH_WORKSPACE_CONFIGURATIONS = ["gong"];
 
+export const GetDefaultTranscriptsConfigurationBodySchema = t.type({
+  provider: acceptableTranscriptProvidersCodec,
```
Comment (flvndvd) [inline front/pages/api/w/[wId]/labs/transcripts/default.ts:15]: Weird that io-ts does not you to use `t.string("gong")`. Are you sure this works?

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx [hunk 3]
```diff
@@ -361,6 +366,32 @@ export default function LabsTranscriptsIndex({
...
+
+    if (!response.ok) {
+      sendNotification({
+        type: "error",
+        title: "Failed to disconnect provider",
+        description:
+          "Could not disconnect from your transcripts provider. Please try again.",
+      });
+    } else {
+      window.location.reload();
```
Comment (flvndvd) [inline front/pages/w/[wId]/assistant/labs/transcripts/index.tsx:389]: 😱 Why not using `mutateTranscriptsConfiguration`?

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx [hunk 4]
```diff
@@ -80,6 +83,8 @@ export default function LabsTranscriptsIndex({
   nangoPublicKey,
 }: InferGetServerSidePropsType<typeof getServerSideProps>) {
   const sendNotification = useContext(SendNotificationsContext);
+  const [isDeleteProviderDialogOpen, setIsDeleteProviderDialogOpen] =
```
Comment (flvndvd) [inline front/pages/w/[wId]/assistant/labs/transcripts/index.tsx:86]:
```suggestion
  const [isDeleteProviderDialogOpened, setIsDeleteProviderDialogOpened] =
```

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx [hunk 5]
```diff
@@ -370,6 +401,20 @@ export default function LabsTranscriptsIndex({
       pageTitle="Dust - Transcripts processing"
       navChildren={<AssistantSidebarMenu owner={owner} />}
     >
+      <Dialog
+        isOpen={isDeleteProviderDialogOpen}
+        title="Disconnect transcripts provider"
+        onValidate={async () =>
+          (await handleDisconnectProvider()) &&
+          setIsDeleteProviderDialogOpen(false)
```
Comment (flvndvd) [inline front/pages/w/[wId]/assistant/labs/transcripts/index.tsx:409]: Let's avoid this and have two distinct lines.

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx (hunk 3 above)
Comment (albandum) [reply front/pages/w/[wId]/assistant/labs/transcripts/index.tsx:389]: Used it but nothing happens :')

File: front/lib/labs/config.ts (hunk 1 above)
Comment (albandum) [reply front/lib/labs/config.ts:11]: Why? Is this a coding style thing on our side?
Makes things more complicated in places like this where I'd have to duplicate line, when typescript handles things nicely for now : https://github.com/dust-tt/dust/pull/5299/files#diff-4f152a814a0e5d1f3e1727fdd4f21be52e3f0f7a49bd5d4f49d912aebe114ea9R158

Comment (flvndvd) [reply front/lib/labs/config.ts:11]: Ok not strongly opinionated against this, it was mainly to have a clear one to one relationship with the environment variables.

File: front/pages/api/w/[wId]/labs/transcripts/default.ts (hunk 2 above)
Comment (flvndvd) [reply front/pages/api/w/[wId]/labs/transcripts/default.ts:15]: IIRC io-ts does not support array out of the box, you need to pass through a union type of string literals.

File: front/pages/w/[wId]/assistant/labs/transcripts/index.tsx (hunk 3 above)
Comment (flvndvd) [reply front/pages/w/[wId]/assistant/labs/transcripts/index.tsx:389]: Then it means that your endpoint should return an empty array in this case 😛

File: front/pages/api/w/[wId]/labs/transcripts/default.ts (hunk 2 above)
Comment (albandum) [reply front/pages/api/w/[wId]/labs/transcripts/default.ts:15]: Yeah that's what I'm doing :) 

```
export const acceptableTranscriptProvidersCodec = t.union([
  t.literal("google_drive"),
  t.literal("gong"),
]);
```

Comment (albandum) [reply front/pages/api/w/[wId]/labs/transcripts/default.ts:15]: Seems like the `PROVIDERS_WITH_WORKSPACE_CONFIGURATIONS` const on the line above confused you, it's not the one used in this line

**Review by flvndvd — APPROVED** [review]
LGTM, left a few last nits.

**Pull request merged**